examples/http/onvif_camera.js

```diff
@@ -0,0 +1,324 @@
+// or
+// profileName = ["All"] to monitor all profiles.
+var profileName = D.getParameter("profileName");
+
+// ONVIF port parameter
+// The port number can be found in the camera's manual or through the camera's admin web app settings. Default is 80.
+var onvifPort = D.getParameter("onvifPort");
+
+var telnetParams = {
+    negotiationMandatory: false,
+    port: 554
+};
+
+// Generate nonce for digest authentication
+var nonce = Math.random().toString(36).substring(2,7);
+var currentDate = new Date();
+
+// The current timestamp in UTC format required for the Created element in the digest authentication
+var created =
+    currentDate.getUTCFullYear() + "-" +
+    ((currentDate.getUTCMonth() + 1) < 10 ? "0" : "") + (currentDate.getUTCMonth() + 1) + "-" +
+    (currentDate.getUTCDate() < 10 ? "0" : "") + currentDate.getUTCDate() + "T" +
+    (currentDate.getUTCHours() < 10 ? "0" : "") + currentDate.getUTCHours() + ":" +
+    (currentDate.getUTCMinutes() < 10 ? "0" : "") + currentDate.getUTCMinutes() + ":" +
+    (currentDate.getUTCSeconds() < 10 ? "0" : "") + currentDate.getUTCSeconds() + "Z";
+
+// Combine nonce, created timestamp, and password to create the string for hashing
+var combinedString =  nonce + created + password;
+// Hash the combined string using SHA1 and encode it in base64 for the Password Digest
+var passwordDigest = D.crypto.hash(combinedString, 'sha1', null, 'base64');
+
+/**
+ * Generates the security envelope for SOAP requests.
+ * @returns {string} The security envelope
+ */
+function getSecurityEnvelope() {
+    return '<wsse:Security>' +
+           '<wsse:UsernameToken>' +
+           '<wsse:Username>' + username + '</wsse:Username>' +
+           '<wsse:Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">' + passwordDigest + '</wsse:Password>' +
+           '<wsse:Nonce>' + D._unsafe.buffer.from(nonce).toString('base64') + '</wsse:Nonce>' +
+           '<wsse:Created>' + created + '</wsse:Created>' +
+           '</wsse:UsernameToken>' +
+           '</wsse:Security>';
+}
+
+/**
+ * Sends a SOAP request and returns a promise with the parsed response.
+ * @param {string} body  The SOAP request body.
+ * @param {function} parseResponse  A function to parse the response body.
+ * @returns A promise that resolves with the parsed response.
+ */
+function sendSoapRequest(body, parseResponse) {
+    var d = D.q.defer();
+    var config = {
+        url: url,
+        username: username,
+        password: password,
+        port: onvifPort,
+        body: body
+    };
+    D.device.http.post(config, function(error, response, body) {
+        if (error) {
+            console.error(error);
+            d.reject(D.errorType.GENERIC_ERROR);
+        } else if (!response) {
+            d.reject(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode == 400) {
+            d.reject(D.errorType.AUTHENTICATION_ERROR);
```

Comment:
change this to 

D.failure(D.errorType.AUTHENTICATION_ERROR)

So that we can capture any authentication error correctly and send it to the callback